csrc/root_domain_map.cpp

```diff
@@ -81,9 +81,13 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseRootDomainMap::map(
   TORCH_INTERNAL_ASSERT(producer_tv_->domain() == producer);
   TORCH_INTERNAL_ASSERT(consumer_tv_->domain() == consumer);
 
+  // In torch.gather, even non-indexed dimensions may have different
+  // extents, whereas in numpy.take_along_axis, they are guaranteed to
+  // be the same
   if (consumer_tv_->definition()->isA<TorchGatherOp>() &&
       consumer_tv_->definition()->as<TorchGatherOp>()->lookupTv() ==
           producerTv() &&
+      !consumer_tv_->definition()->as<TorchGatherOp>()->isTakeAlongAxis() &&
```

Comment:
This is an important difference between `torch.gather` and `numpy.take_along_axis`